src/libraries/System.Private.CoreLib/src/System/Runtime/CompilerServices/RuntimeHelpers.MemOps.cs

```diff
@@ -0,0 +1,537 @@
+                }
+
+                // It's possible that some small buffer remains to be populated - something that won't
+                // fit an entire vector's worth of data. Instead of falling back to a loop, we'll write
+                // a vector at the very end of the buffer. This may involve overwriting previously
+                // populated data, which is fine since we're splatting the same value for all entries.
+                // There's no need to perform a length check here because we already performed this
+                // check before entering the vectorized code path.
+
+                Unsafe.WriteUnaligned(ref Unsafe.AddByteOffset(ref dest, numElements - (nuint)Vector<byte>.Count), vector);
+
+                // And we're done!
+                return;
+            }
+
+        CannotVectorize:
+            // If we reached this point, we cannot vectorize this T, or there are too few
+            // elements for us to vectorize. Fall back to an unrolled loop.
+
+            nuint i = 0;
+
+            // Write 8 elements at a time
+            if (numElements >= 8)
+            {
+                nuint stopLoopAtOffset = numElements & ~(nuint)7;
+                do
+                {
+                    Unsafe.Add(ref dest, (nint)i + 0) = value;
+                    Unsafe.Add(ref dest, (nint)i + 1) = value;
+                    Unsafe.Add(ref dest, (nint)i + 2) = value;
+                    Unsafe.Add(ref dest, (nint)i + 3) = value;
+                    Unsafe.Add(ref dest, (nint)i + 4) = value;
+                    Unsafe.Add(ref dest, (nint)i + 5) = value;
+                    Unsafe.Add(ref dest, (nint)i + 6) = value;
+                    Unsafe.Add(ref dest, (nint)i + 7) = value;
+                } while ((i += 8) < stopLoopAtOffset);
+            }
+
+            // Write next 4 elements if needed
+            if ((numElements & 4) != 0)
+            {
+                Unsafe.Add(ref dest, (nint)i + 0) = value;
+                Unsafe.Add(ref dest, (nint)i + 1) = value;
+                Unsafe.Add(ref dest, (nint)i + 2) = value;
+                Unsafe.Add(ref dest, (nint)i + 3) = value;
+                i += 4;
+            }
+
+            // Write next 2 elements if needed
+            if ((numElements & 2) != 0)
+            {
+                Unsafe.Add(ref dest, (nint)i + 0) = value;
+                Unsafe.Add(ref dest, (nint)i + 1) = value;
+                i += 2;
+            }
+
+            // Write final element if needed
+            if ((numElements & 1) != 0)
+            {
+                Unsafe.Add(ref dest, (nint)i) = value;
+            }
+        }
+
+#if NATIVEAOT
+        [System.Runtime.RuntimeExport("RhRuntimeHelpers_MemZero")]
+#endif
+        private static unsafe void MemZero(ref byte b, nuint byteLength)
+        {
+            if (byteLength == 0)
+                return;
+
+            // Implicit nullcheck
+            _ = Unsafe.ReadUnaligned<byte>(ref b);
```

Comment:
This can be moved to be after the `PInvoke:` label.

Reply:
Once the null check is moved to be after the `PInvoke:` label, there should be no material difference between this and ClearWithoutReferences - can we have just one implementation?